x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ltr/LearningToRankRescorerBuilder.java

```diff
@@ -33,6 +33,7 @@
 public class LearningToRankRescorerBuilder extends RescorerBuilder<LearningToRankRescorerBuilder> {
 
     public static final String NAME = "learning_to_rank";
+    public static final int DEFAULT_WINDOW_SIZE = 1000;
```

Comment:
`1000` is too big imo. I'd prefer that we document the limitation clearly for pagination since no matter what value we pick here this will be surprising for users.

Reply:
1k is pretty standard practice. These models are very fast, nothing like NLP models, so 1k is usually quite reasonable. If we want to be cautious about a default, 100 would be the absolute smallest I would suggest. 200 or 500 would definitely be better. We can also do some quick benchmarking on a dataset to get some basic performance values if that helps us make a decision.

Reply:
IMO the ideal would be to make this parameter non-optional

Reply:
Question, how hard would it be to make this default to `min(window_size, 1000)`?

Reply:
> make this parameter non-optional
@jimczi ++ As a first release as well this makes sense and we can always make it optional later.

> how hard would it be to
@kderusso You mean make the actual window size `min(window_size, 1000)`? I think this would lead to unexpected behaviour. Just making this parameter required is definitely the clearest thing we can do. If you're doing LTR, you really do need to pick your window size.

Reply:
The new version of the code make the window_size required for LTR rescorer.
I let you review this new version.